src/app/forms/custom-sections/letters/letters.component.ts

```diff
@@ -75,9 +78,16 @@ export class LettersComponent implements OnInit, OnDestroy, OnChanges {
   }
 
   get eligibleForLetter(): boolean {
-    const currentTechRecord = this.techRecord?.techRecord_statusCode === StatusCodes.CURRENT;
-
-    return this.correctApprovalType && currentTechRecord && !this.isEditing;
+    let hasCurrentInHistory = false;
+    this.techRecordService.techRecordHistory$.subscribe((historyArray: TechRecordSearchSchema[] | undefined) => {
+      historyArray?.forEach((history: TechRecordSearchSchema) => {
+        if (history.techRecord_statusCode === StatusCodes.CURRENT
+          && this.techRecord?.techRecord_statusCode === StatusCodes.PROVISIONAL) hasCurrentInHistory = true;
+      });
+    });
+    const currentTechRecord = this.techRecord?.techRecord_statusCode === StatusCodes.CURRENT
+    || this.techRecord?.techRecord_statusCode === StatusCodes.PROVISIONAL;
+    return this.correctApprovalType && currentTechRecord && !this.isEditing && !hasCurrentInHistory;
```

Comment:
same for this, it either needs to be split out into separate functions which give an indication as to what is being done, or comments need to be added to make it more understandable

Reply:
all of this needs unit testing too

Reply:
Yup, I think so. I will make it more readable.